nixos/modules/services/web-apps/pixivfe.nix

```
@@ -0,0 +1,126 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.pixivfe;
+  settingsFormat = pkgs.formats.yaml { };
+in
+{
+  options.services.pixivfe = {
+    enable = lib.mkEnableOption "PixivFE, a privacy respecting frontend for Pixiv";
+
+    package = lib.mkPackageOption pkgs "pixivfe" { };
+
+    openFirewall = lib.mkEnableOption "open ports in the firewall needed for the daemon to function";
+
+    settings = lib.mkOption {
+      type = lib.types.nullOr (
+        lib.types.submodule {
+          freeformType = settingsFormat.type;
+        }
+      );
+      default = { };
+      example = lib.literalExpression ''
+        {
+          port = "8282";
+          token = "123456_AaBbccDDeeFFggHHIiJjkkllmMnnooPP";
+        };
+      '';
+      description = ''
+        Additional configuration for PixivFE, see
+        <https://pixivfe-docs.pages.dev/hosting/configuration-options/> for supported values.
+        For secrets use `EnvironmentFile` option instead.
+      '';
+    };
+
+    EnvironmentFile = lib.mkOption {
+      type = lib.types.nullOr lib.types.str;
+      default = null;
+      example = lib.literalExpression ''
+        /run/secrets/environment
+      '';
```

```suggestion
      example = lib.literalExpression "/run/secrets/environment";
```
this is not a multiline string